### PR TITLE
feat: improvements to the issue/share flows

### PR DESCRIPTION
**Pull request by janrtvld**
Feedback still welcome. The text will probably also still change.

https://github.com/user-attachments/assets/f44609d6-d068-49b5-8991-87d51a274069

### CODE REVIEW

File: apps/easypid/src/features/receive/slides/CredentialRetrievalSlide.tsx [hunk 1]
```diff
@@ -25,19 +33,44 @@ import {
...
+  attributes: Record<string, unknown>
   display: CredentialDisplay
   isCompleted: boolean
+  onAccept: () => Promise<void>
   onGoToWallet: () => void
 }
 
-export const CredentialRetrievalSlide = ({ display, isCompleted, onGoToWallet }: CredentialRetrievalSlideProps) => {
-  const { completeProgressBar } = useWizard()
+export const CredentialRetrievalSlide = ({
```
Comment (janrtvld) [inline apps/easypid/src/features/receive/slides/CredentialRetrievalSlide.tsx:43]: I know this file is a bit of a mess with all the states so I'm thinking about better abstractions for the animations.

But will prob be after the deadline

File: apps/easypid/src/features/receive/slides/CredentialCardSlide.tsx [hunk 2]
```diff
@@ -1,46 +1,64 @@
...
+  switch (type) {
+    case 'presentation':
+      return {
+        title: 'Share who you are',
+        subtitle: `To receive this card from ${issuerName}, you need to share your identity.`,
+      }
+    case 'pin':
+      return {
+        title: 'Card offered',
+        subtitle: `To receive this card from ${issuerName}, you need to enter your PIN.`,
```
Comment (TimoGlastra) [inline apps/easypid/src/features/receive/slides/CredentialCardSlide.tsx:20]: I think it's good to keep it as a transaction PIN or something, to not confuse with other PINs used?

```suggestion
        subtitle: `To receive this card from ${issuerName}, you need to enter a transaction PIN.`,
```

File: apps/easypid/src/features/receive/slides/CredentialCardSlide.tsx [hunk 3]
```diff
@@ -1,46 +1,64 @@
...
-  onContinue: () => void
 }
 
-export const CredentialCardSlide = ({ display, onContinue }: CredentialCardSlideProps) => {
+const getContentType = (type: 'presentation' | 'pin' | 'noAuth', issuerName: string) => {
+  switch (type) {
+    case 'presentation':
+      return {
+        title: 'Share who you are',
+        subtitle: `To receive this card from ${issuerName}, you need to share your identity.`,
```
Comment (TimoGlastra) [inline apps/easypid/src/features/receive/slides/CredentialCardSlide.tsx:15]: It's not neccesarily that you share who you are. Maybe you'll need to share something about your dog, who knows. 

cc @anagoessens feel free to make suggestions here.

```suggestion
        title: 'Sharing required',
        subtitle: `To receive this card from ${issuerName}, you need to share existing cards in your wallet.`,
```

File: apps/easypid/src/features/receive/slides/CredentialCardSlide.tsx [hunk 4]
```diff
@@ -1,46 +1,64 @@
...
+      }
+    case 'pin':
+      return {
+        title: 'Card offered',
+        subtitle: `To receive this card from ${issuerName}, you need to enter your PIN.`,
+      }
+    default:
+      return {
+        title: 'Card offered',
+        subtitle: `${issuerName} wants to give you the following card:`,
```
Comment (TimoGlastra) [inline apps/easypid/src/features/receive/slides/CredentialCardSlide.tsx:25]: give is maybe not hte right word, is issue too technical?

```suggestion
        subtitle: `${issuerName} wants to issue you the following card:`,
```

**Review by TimoGlastra — APPROVED** [review]
nice

**Pull request merged**